core/src/main/java/rdx/works/core/sargon/MnemonicWithPassphraseExtensions.kt

```diff
@@ -0,0 +1,106 @@
+package rdx.works.core.sargon
+
+import com.radixdlt.bip39.generateMnemonic
+import com.radixdlt.bip39.model.MnemonicWords
+import com.radixdlt.bip39.toSeed
+import com.radixdlt.bip39.wordlists.WORDLIST_ENGLISH
+import com.radixdlt.crypto.ec.EllipticCurveType
+import com.radixdlt.crypto.getCompressedPublicKey
+import com.radixdlt.extensions.removeLeadingZero
+import com.radixdlt.sargon.BIP39Passphrase
+import com.radixdlt.sargon.Bip39WordCount
+import com.radixdlt.sargon.DerivationPath
+import com.radixdlt.sargon.FactorSource
+import com.radixdlt.sargon.FactorSourceId
+import com.radixdlt.sargon.FactorSourceIdFromHash
+import com.radixdlt.sargon.FactorSourceKind
+import com.radixdlt.sargon.HierarchicalDeterministicPublicKey
+import com.radixdlt.sargon.Mnemonic
+import com.radixdlt.sargon.MnemonicWithPassphrase
+import com.radixdlt.sargon.PublicKey
+import com.radixdlt.sargon.Slip10Curve
+import com.radixdlt.sargon.extensions.asGeneral
+import com.radixdlt.sargon.extensions.bytes
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.string
+import com.radixdlt.sargon.extensions.toBagOfBytes
+import com.radixdlt.slip10.model.ExtendedKey
+import com.radixdlt.slip10.toKey
+import rdx.works.core.crypto.PrivateKey
+import rdx.works.core.crypto.PrivateKey.Companion.toPrivateKey
+import rdx.works.core.hash
+
+fun MnemonicWithPassphrase.Companion.init(phrase: String) = MnemonicWithPassphrase(
+    mnemonic = Mnemonic.init(phrase = phrase),
+    passphrase = BIP39Passphrase()
+)
+
+/**
+ * Generates a mnemonic based on the [WORDLIST_ENGLISH]. Used only when no mnemonic
+ * exists.
+ */
+fun MnemonicWithPassphrase.Companion.generate(
+    entropyStrength: Int
+): MnemonicWithPassphrase = MnemonicWithPassphrase(
+    mnemonic = Mnemonic.init(
+        phrase = generateMnemonic(
+            strength = entropyStrength,
+            wordList = WORDLIST_ENGLISH
+        )
+    ),
+    passphrase = BIP39Passphrase()
+)
+
+fun MnemonicWithPassphrase.toFactorSourceId(
+    curve: Slip10Curve = Slip10Curve.CURVE25519
+): FactorSourceId.Hash = FactorSourceIdFromHash(
+    kind = FactorSourceKind.DEVICE,
+    body = toExtendedKey(
+        curve = curve,
+        derivationPath = DerivationPath.getIdPath()
+    ).keyPair.getCompressedPublicKey().removeLeadingZero().hash().bytes
+).asGeneral()
+
+fun MnemonicWithPassphrase.derivePrivateKey(
+    hdPublicKey: HierarchicalDeterministicPublicKey
+): PrivateKey = toExtendedKey(
+    curve = when (hdPublicKey.publicKey) {
+        is PublicKey.Ed25519 -> Slip10Curve.CURVE25519
+        is PublicKey.Secp256k1 -> Slip10Curve.SECP256K1
+    },
+    derivationPath = hdPublicKey.derivationPath
+).keyPair.toPrivateKey()
+
+fun MnemonicWithPassphrase.derivePublicKey(
```

Comment:
this still uses slip10 lib? I thought this will happen in sargon

Reply:
PrivateKey and Slip10 will be removed in the next PR.